@swinbank updates to DM-19698

jdswinbank opened the pull request:
@gcomoretto  — as discussed, these are my suggested changes to LDM-672.

I am very happy with the document in its current state: thank you for doing a good job. Almost all of my changes are cosmetic or wording improvements and minor fixes.

The one thing I didn't understand (and that we briefly discussed on Slack) is the relationship of the stakeholders described in §2.1 and §2.6 of the document. I understood the full list of stakeholders being discussed to be:

- The LSST Construction Project, including commissioning and science validation
- The Science Community
- Data Processing in Operations
- Other uses in the operational observatory which are not directly generating data products
- The Science Platform and other user-facing tools

But then... §2.6 came along, and I don't understand what it means! :-) I did not actually remove it as I suggested earlier, but I would appreciate it if you can clarify what community it is supposed to represent.

When that has been resolved, I think it would be appropriate to baseline this document.

jdswinbank committed:
Fix TEXMFHOME.

jdswinbank committed:
Remove duplicate material.

jdswinbank committed:
Minor rewording of requirements section.

jdswinbank committed:
Tweaks to consolidated requiremnents.

jdswinbank committed:
Tweak policy wording.

jdswinbank committed:
Reword applicability.

jdswinbank committed:
Polish release policy.

gcomoretto commented on policy.tex:
```diff
@@ -1,67 +1,67 @@
-\item if it is a time-based release: specify the timing when the release shall be done (for example, every 6 months)
-\item if it is a functionality based release: specify which functionalities shall be included in the release
+\item when the release is expected;
+\item the specify the corresponding milestone or RFC;
```
'the specify the corresponding...' doesn't sound correct to me... but I may be wrong.

jdswinbank replied:
It's not. Good catch!

gcomoretto commented on requirements.tex:
```diff
@@ -2,73 +2,78 @@ \section{Stakeholders Requirements} \label{sec:reqs}
+\subsection{Release Requirements for the \gls{LSST} Construction Project} \label{sec:lsstreqs}
 
-The main reason for DM to do software releases is to fulfill the project outcome: provide the Data Products.
+The main reason for DM to make software releases during construction is to support development and commissioning activities undertaken within DM and elsewhere, and to verify progress towards the ultimate completion of the system.
```
I disagree with this. Supporting development and commissioning is not the (final) scope of the project.
Probably it is just a matter of phrasing.

I would say: 
- the project needs to produce Data Products (this is the outcome).
- the production of the project Data Products has to be controlled and reproducible, therefore
- the Data Products need to be obtained using released software (in operations).
- this implies that DM has to be able, already during the contruction phase, to provide released software to use for the consolidation and preparation of operations.

jdswinbank replied:
That is explicitly covered by “support commissioning activities... and verify progress towards the ultimate completion of the system”. I don't understand what you want to add.

gcomoretto replied:
Stakeholder in 2.1 was meant to be general: "Release Requirements for the LSST Project".
And the focus has to stay on the main outcome of the project: the data releases.

Your rephrasing makes this section be the same as 2.6, that was originally meant for commissioning and verification activities before 2.1 was added.

I understand now that my original sentence is including parts that are already in the other sections.

I also disagree that we are doing releases to support development.
We are making available weekly and daily builds to support developments, but those are not releases.

gcomoretto commented on requirements.tex:
```diff
@@ -2,73 +2,78 @@ \section{Stakeholders Requirements} \label{sec:reqs}
 
 
-\subsection{Release Requirements for Infrastructural Software} \label{sec:infreqs}
+\subsection{Release Requirements for User-Facing Infrastructure} \label{sec:infreqs}
```
It is not just user-oriented software, LSP is is just an example. What about Qserv?

jdswinbank replied:
Hmm, interesting point.

I think Qserv *is* “user-facing infrastructure”, in that it is infrastructure which is designed for serving data to users, as opposed to infrastructure for operating the observatory (which is defined in the previous section).

If you have a better term than “user-facing infrastructure”, that would be great! But we can't just say “infrastructure” without it being understood to mean things like the Telescope Control System.

gcomoretto replied:
Fine. I suppose that adding Software at the end is making the subsection title too long...

gcomoretto reviewed (changes requested):
Despite this is a DM policy, I think we shall keep in mind what is the final goal of the project.
Of course, the DM releases will enable commissioning activities and construction in general. 
But the DM releases shall be done in order to permit the LSST Data Products to be produced.
The project requirement should be strict: only officially released software shall be used to produce LSST Data Products.

Since verification and commissioning may use unreleased software, having that activities as a different stakeholder (in 2.6) is recommendable: it relaxes the above requirement.

Section 2.1 has been added only recently, the text may need some cleaning in order to avoid overlapping:
- 2.1 shall cover what is the final goal of the project
- 2.6 is covering in reality what is the construction phase, and should probably be moved in the second position.

jdswinbank commented:
I understand requirements like “only officially released software shall be used to produce LSST Data Products” to be covered by the “Release Requirements for Data Processing in Operations” section.

Could you explain why you think that section is inadequate?

gcomoretto reviewed (commented):
This section (2.3) is meant to cover requirements from data processing centers.

However saying that we are doing data processing in operation, or construction is not sufficient.
Why are we doing that? Because we want to release Data Products.

I think we need to have a general stakeholder, that justify why we are doing all this: the LSST Project.

My error has been to add it, without paying attention to not overlap with the other stakeholders already defined, that in reality are a consequence of 2.1.

We should keep 2.1 generic, and rephrase the text in order to avoid overlapping with other stakeholders, and avoid confusion.

The other option I see now, is to remove this general stakeholder, as you have done, and include the final scope of the project in section 2 introduction. In this case, section 2.6 becomes obsolete.

jdswinbank commented:
Hi @gcomoretto — having read your comments, I'm still a bit confused about who all the various stakeholders you're considering are and in what ways they are distinct from each other.

I suggest either:

- You try redrafting to reduce overlap, as you suggest above; I'll take another look at that version, and see if I can then understand what you're aiming for; or
- We can chat tomorrow to see if we can sort it out by voice.

I'm happy with whichever you prefer.

jdswinbank committed:
Fix typo.

jdswinbank committed:
Clean up requirements table.

jdswinbank committed:
Add me as author.

jdswinbank commented:
@gcomoretto — I've rebased my PR along the lines we discussed earlier. Please take a look and let me know what you think.

One thing I wanted to mention: I changed the wording in §2.1 slight to reflect the fact that “Data Management” no longer exists as a subsystem during operations, and that all construction milestones will already have been accomplished by definition. Obviously, it makes no sense for this document to say that a non-existent organization will release software to meet milestones which have already been completed! 😄 I hope you find my new wording acceptable — I think it sticks to the spirit of what you were aiming for.

I also took the liberty of adding myself to the author list, since I have (by lines of code, at least) contributed more than any of the other authors except you. If you don't think this is appropriate, please feel free to remove me (I won't be offended!!).

gcomoretto commented on requirements.tex:
```diff
@@ -2,86 +2,89 @@ \section{Stakeholders Requirements} \label{sec:reqs}
 
 
-\subsection{Release Requirements for the \gls{LSST} Science Community} \label{sec:comreqs}
+\subsection{Release Requirements for the Interacting with the Scientific Community} \label{sec:comreqs}
```
Should be here ".. for Ingreracting with..." ?

jdswinbank replied:
“Integrating”? I did mean “interacting”, but I think “integrating” would work just about as well, so I've no objection to changing this if you want.

gcomoretto replied:
Yes.. sorry. I think ".. for the Interacting..." should be just "... for Interacting ..."

jdswinbank replied:
Oh, you are quite right, of course. :-)

gcomoretto commented on requirements.tex:
```diff
@@ -143,62 +145,64 @@ \subsection{Stakeholders Requirements} \label{sec:stakeholdersreqs}
-In both cases, releases can be tight to project milestones.
-Additional releases can be requested to the DM-CCB using the RFC mechanism.
+In both cases, releases may be associated with project milestones.
+Additional releases may be made available upon request to the DM-CCB using the “RFC” mechanism as described in \citeds{DMTN-106}.
```
The RFC approach is described in this document, section 6. DMTN-106 may give technicalities on how to do path releases, and DevGuide on how to backport, but how to request a new release via RFC is described here.

jdswinbank replied:
ok

gcomoretto reviewed (approved):
Thank you John for your inputs, and for reformatting the summary table in 3.3.

I fully agree you are added in the authors list. I forgot to mention this morning.

I think we are ready to merge and do the RFC.

jdswinbank commented:
Great! I have pushed a new reference for the “RFC” text you queried above; when Travis finishes, I will merge this with your ticket branch, and leave you to finish the process. :-)

jdswinbank commented:
All done!
